include/aws/mqtt/private/client_impl_shared.h

```diff
@@ -132,6 +134,28 @@ struct aws_mqtt_client_connection {
     void *impl;
 };
 
+struct aws_mqtt_iot_sdk_metrics_storage {
```

Comment:
Probably better off in a separate file.  this file is really intended for 311 (or the adapter)